assets/wizards/newspack/views/settings/connections/google-oauth.tsx

```diff
@@ -0,0 +1,170 @@
+/**
+ * Settings Wizard: Connections > Google OAuth
+ */
+
+/**
+ * WordPress dependencies.
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies.
+ */
+import { Button } from '../../../../../components/src';
+import WizardsActionCard from '../../../../wizards-action-card';
+import { useWizardApiFetch } from '../../../../hooks/use-wizard-api-fetch';
+import { WizardError, WIZARD_ERROR_MESSAGES } from '../../../../errors';
+
+function getURLParams() {
+	const searchParams = new URLSearchParams( window.location.search );
+	const params: { [ key: string ]: string } = {};
+	for ( const [ key, value ] of searchParams.entries() ) {
+		params[ key ] = value;
+	}
+	return params;
+}
+
+function GoogleOAuth( {
+	onSuccess,
+	isOnboarding,
+}: {
+	onInit?: ( str: Error | null ) => void;
+	onSuccess?: ( arg: OAuthData ) => void;
+	isOnboarding?: ( str: string ) => void;
+} ) {
+	const [ authState, setAuthState ] = useState< OAuthData >( {} );
+
+	const userBasicInfo = authState?.user_basic_info;
+	const isConnected = Boolean( userBasicInfo && userBasicInfo.email );
+
+	const {
+		setError,
+		errorMessage,
+		wizardApiFetch,
+		isFetching: inFlight,
+	} = useWizardApiFetch( '/newspack-settings/connections/apis/google-oauth' );
+
+	// We only want to autofetch the current auth state if we're not onboarding.
+	useEffect( () => {
+		if ( ! isOnboarding ) {
+			getCurrentAuth();
+		}
+	}, [] );
+
+	useEffect( () => {
+		if ( isConnected && userBasicInfo && ! userBasicInfo.has_refresh_token ) {
+			setError(
+				new WizardError(
+					WIZARD_ERROR_MESSAGES.GOOGLEOAUTH_REFRESH_TOKEN_EXPIRED,
+					'googleoauth_refresh_token_expired'
+				)
+			);
+		}
+	}, [ isConnected ] );
+
+	function getCurrentAuth() {
+		const params = getURLParams();
+		if ( ! params.access_token ) {
+			wizardApiFetch< OAuthData >(
+				{
+					path: '/newspack/v1/oauth/google',
+				},
+				{
+					onSuccess( data ) {
+						setAuthState( data );
+						if ( data?.user_basic_info ) {
+							if ( typeof onSuccess === 'function' ) {
+								onSuccess( data );
+							}
+						}
+					},
+				}
+			);
+		}
+	}
+
+	function openAuth() {
+		let authWindow: Window | null = null;
+		wizardApiFetch< string >(
+			{
+				path: '/newspack/v1/oauth/google/start',
+			},
+			{
+				onSuccess( url ) {
+					authWindow = window.open(
+						'about:blank',
```

Comment:
We should be able to just open with `url` directly here. I don't think there is a reason to wait until the popup opens to update the location, is there?

Reply:
@chickenn00dle have updated.